flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/DataStreamCsvITCase.java

```diff
@@ -82,41 +102,86 @@ public void testBoundedTextFileSourceWithJackson() throws Exception {
+        final File outDir = TMP_FOLDER.newFolder();
+
+        StreamingFileSink<CityPojo> sink =
+                StreamingFileSink.forBulkFormat(
+                                new Path(outDir.toURI()), new TestCsvBulkWriterFactory())
+                        .withBucketAssigner(new BasePathBucketAssigner<>())
+                        .withBucketCheckInterval(10L)
+                        .withRollingPolicy(build())
+                        .build();
+
+        OneInputStreamOperatorTestHarness<CityPojo, Object> testHarness =
+                new OneInputStreamOperatorTestHarness<>(new StreamSink<>(sink), 1, 1, 0);
+
+        testHarness.setup();
+        testHarness.open();
+
+        testHarness.processElement(new StreamRecord<>(pojos[0], 1L));
+        testHarness.processElement(new StreamRecord<>(pojos[1], 1L));
+        testHarness.processElement(new StreamRecord<>(pojos[2], 1L));
+        testHarness.snapshot(1L, 1L);
+        testHarness.notifyOfCompletedCheckpoint(2L);
+
+        Map<File, String> contents = getFileContentByPath(outDir);
+        Map.Entry<File, String> onlyEntry = contents.entrySet().iterator().next();
+        String[] result = onlyEntry.getValue().split("\n");
+
+        assertThat(result, is(CSV_LINES));
+        //        assertThat(result, arrayContaining(CSV_LINES));
+    }
+
+    static Map<File, String> getFileContentByPath(File directory) throws IOException {
+        Map<File, String> contents = new HashMap<>(4);
+
+        final Collection<File> filesInBucket = FileUtils.listFiles(directory, null, true);
+        for (File file : filesInBucket) {
+            contents.put(file, FileUtils.readFileToString(file));
+        }
+        return contents;
+    }
+
+    static final class TestCsvBulkWriterFactory implements BulkWriter.Factory<CityPojo> {
+
+        private static final long serialVersionUID = 1L;
+
+        @Override
+        public BulkWriter<CityPojo> create(FSDataOutputStream out) {
+
+            final ObjectMapper objectMapper = new ObjectMapper();
+            final CsvMapper csvMapper = new CsvMapper();
+            final CsvSchema schema = csvMapper.schemaFor(CityPojo.class).withoutQuoteChar();
+
+            PojoConverter.ConverterContext converterContext =
+                    new PojoConverter.ConverterContext(objectMapper);
+            final PojoConverter converter = createConverter();
+
+            return CsvBulkWriter.forSchema(csvMapper, schema, converter, converterContext, out);
+        }
+    }
+
+    interface PojoConverter extends Converter<CityPojo, JsonNode, PojoConverter.ConverterContext> {
+
+        class ConverterContext implements Serializable {
+            ObjectMapper mapper;
+
+            public ConverterContext(ObjectMapper csvMapper) {
+                this.mapper = csvMapper;
+            }
+        }
+    }
+
+    public static PojoConverter createConverter() {
+        return (pojo, context) -> context.mapper.valueToTree(pojo);
```

Comment:
Why do we need a special PojoConverter? On reading side, we just do `(value, context) -> value` for POJO, right? Shouldn't the converter already receive the mapped value?

Reply:
Added native POJOs support in CsvBulkWriter.